### PR TITLE
FEM Lab: use FEM preview links, etc

### LINKED ISSUE #6049: FEM Lab: Nov 2021 misc TODO
## Misc ToDo

New behaviour:
- [ ] on FEM-Lab, `workflow.configuration.persist_annotations === undefined` should imply **true**.
  - On PFE-Lab, undefined implies false
  - See zooniverse/front-end-monorepo#2524
- [ ] NEW: on FEM-Lab, `workflow.configuration.hide_classification_summaries ===undefined` should also imply **true**.
- [ ] on FEM-lab, "View Project" and "Test this Workflow" should both point to the FEM preview URL.
  - ~~Note that we want to use `fe-project.zooniverse.org` points to the _production-release_ version of FEM...~~
  - ~~...whereas `frontend.preview.zooniverse.org` points to the _master_ version of FEM, which has not-yet-deployed changes.~~
  - EDIT: use `frontend.preview.zooniverse.org`
- [ ] FEM-Lab should have some visual indicators to separate it from PFE-Lab

### PR DESCRIPTION
**Pull request by shaunanoordin**
## PR Overview

Closes #6049

This PR updates the FEM Lab with...
- [x] preview links that point to `frontend.preview.zooniverse.org`
- [x] `persist_annotations` or `hide_classification_summaries` checkboxes that are ticked by default (i.e. undefined implies true)

### Status

WIP

Defers #6051 to another PR.

### CODE REVIEW

**Comment by shaunanoordin** [comment]
## PR Update

This PR is now ready for review

### Testing

Setup:
- Test URL: `https://pr-6053.pfe-preview.zooniverse.org/lab/(ANY-PROJECT)/workflows/(ANY-WORKFLOW)` Project Builder -> Workflow Editor page
- You can force the FEM or PFE mode of the Project Builder by using `?pfeLab=true` or `femLab=true`
  - e.g. https://pr-6053.pfe-preview.zooniverse.org/lab/1873/workflows/3532?femLab=true

What you're testing for:
- In FEM Lab mode...
  - The "View Project" and "Test this Workflow" links should link to `frontend.preview.zooniverse.org/...`
  - "Persist Annotations" and "Hide classification summaries" checkboxes should default to ticked (assuming you've never changed these settings for this workflow before) AND disabled (i.e. you can't change it)
- In PFE mode...
  - The "View Project" and "Test this Workflow" links should link to `pr-6053.pfe-preview.zooniverse.org/...`

File: app/pages/lab/project.cjsx [hunk 1]
```diff
@@ -53,13 +54,25 @@ EditProjectPage = createReactClass
     linkParams =
       projectID: @props.project.id
 
+    thisProjectUsesFEM = isThisProjectUsingFEMLab @props.project, @props.location
+
+    projectLink = "/projects/#{@props.project.slug}"
+    if thisProjectUsesFEM then projectLink = "#{FEM_LAB_PREVIEW_HOST}#{projectLink}"
```
Comment (eatyourgreens) [inline app/pages/lab/project.cjsx:60]: This is going to need to add `?env=staging` on staging.
eg. here the project preview link gives a 404 error.
https://pr-6053.pfe-preview.zooniverse.org/lab/908?femLab=true

File: app/pages/lab-fem/workflow.cjsx [hunk 2]
```diff
@@ -47,9 +48,14 @@ EditWorkflowPage = createReactClass
   workflowLink: ->
     [owner, name] = @props.project.slug.split('/')
     usingTranscriptionTask = Object.keys(@props.workflow.tasks).some((taskKey) => @props.workflow.tasks[taskKey].type is 'transcription')
-    if @canUseTask(@props.project, "transcription-task") and usingTranscriptionTask
+    if isThisProjectUsingFEMLab(@props.project, @props.location)
+      return "#{FEM_LAB_PREVIEW_HOST}/projects/#{owner}/#{name}/classify/workflow/#{@props.workflow.id}"
```
Comment (eatyourgreens) [inline app/pages/lab-fem/workflow.cjsx:52]: Same comment about adding `?env=staging` for staging projects here too.

**Review by eatyourgreens — COMMENTED** [review]
A couple of comments about preview links being broken on the staging API. Otherwise, this looks good for production projects.

File: app/pages/lab/project.cjsx (hunk 1 above)
Comment (eatyourgreens) [reply app/pages/lab/project.cjsx:60]: I guess this is complicated by the fact that `*.pfe-preview.zooniverse.org` uses the staging API, but `frontend.preview.zooniverse.org` uses the production API.

**Review by eatyourgreens — APPROVED** [review]
Looks good to me. Just a couple of comments about how this works with the staging API, and I'm not sure about `location.query`. It works, but that's not a standard property of the Location interface.
https://developer.mozilla.org/en-US/docs/Web/API/Location

**Comment by shaunanoordin** [comment]
Thanks Jim!

- noted that the preview links on the Workflow Editor don't work on PFE staging.
  - I initially opted not use Sarah's `env = process.env.NODE_ENV` pattern because I felt it made the links on production look a bit worse (i.e. explicitly stating ?env=production), but you're right that this shouldn't break the staging links. Gonna get in a patch for this. 🛠️ 
- re: `location`, I was passing in the 'location' data from the react-router since it was there. I can change this to e.g. `(new URL(window.location)).searchParams.get('femLab')` if anyone prefers, I don't feel strongly either way. 👌

**Comment by eatyourgreens** [comment]
`NODE_ENV` controls the webpack build mode, so 'staging' isn't really a valid value. Webpack build modes are `none | development | production`.
https://webpack.js.org/configuration/mode/

In the monorepo, we've used `APP_ENV` and `PANOPTES_ENV` to indicate the deployment environment and the API but that would probably be a big change for PFE, when we want to build a production build that uses the staging API.

**Pull request merged**